internal/custom_actions/custom_actions.lua

```diff
@@ -57,4 +60,90 @@ function custom_actions.splitItemsAtTimeSelection()
   reaper.Main_OnCommand(SplitAtTimeSelection, 0)
 end
 
+function custom_actions.updatePrefixOfSelectedTracks() trackUpdateName(1) end
+function custom_actions.updateNameOfSelectedTracks() trackUpdateName(0) end
+
+-- mv to util/track.lua
+function trackUpdateName(set_prefix)
+  log.clear()
+  local num_sel = reaper.CountSelectedTracks(0)
+  local _, new_name_string = reaper.GetUserInputs("Change track name", 1, "Track name:", "")
+  if num_sel == 0 then return end
+
+  if num_sel > 0 then
+    for i = 1, num_sel do
+      local tr = reaper.GetSelectedTrack(0, i - 1)
+      local ret, old_name_full = reaper.GetTrackName(tr)
+      local s, e = string.find(old_name_full, config.name_prefix_match_str)
+      if s == nil then s = 0; e = 0 end
+      local old_prefix = string.sub(old_name_full, s, e)
+      local old_name = string.sub(old_name_full, e + 1)
+
+      local new_name_full
+      if set_prefix == 1 then
+        new_name_full = new_name_string .. old_name
+      else
+        new_name_full = old_prefix .. new_name_string
+      end
+      local _, str = reaper.GetSetMediaTrackInfo_String(tr, "P_NAME", new_name_full, 1);
+    end
+    return
+  end
+end
+
+function updateMidiPreProcessorByInputDevice(tr)
+  local tr_rec_in = reaper.GetMediaTrackInfo_Value(tr, 'I_RECINPUT')
+  local midi_device_offset = 4096
+  local device_mask = 2016
+  local dev_id = ((tr_rec_in - midi_device_offset) & device_mask) >> 5
+  local retval, nameout = reaper.GetMIDIInputName( dev_id, '' )
+
+  local enabled_device
+  for k,device_str in pairs(io.midi) do
+    if nameout:lower():match(device_str:lower()) then enabled_device = device_str end
+  end
+
+  if enabled_device == nil then return end
+  if enabled_device == io.midi.vkb then
+    fx.setParamForFxAtIndex(tr, 0, 1, 0, true) -- set device
```

Comment:
instead of comments to describe the functions here, I would create a variable or function to explain what is happening